### DISCUSSION TITLE
How is the aggregate score calculated?

### DISCUSSION THREAD

**Discussion by soren5** [trane-project/trane#173] [Q&A] [answered]
I am very interested in this project and decided to try it out in some of my own hobbies.
I want to build some visualization for my progress and I noticed that `trane scores` calculates an "Aggregate Score".
I am trying to find how this value is calculated but I am unfamiliar with rust so it is difficult for me to find the related code.

How is the aggregate score calculated?

**Comment by martinmr** [comment, marked as answer]
The code is here: https://github.com/trane-project/trane/blob/master/src/scorer.rs#L41

The short answer is that the final score is a weighted average of the scores. The weight is based on how recent the score is. And each score is adjusted depending on how recent it is and the position within the input array. The score of a trial is lowered with each passing day to simulate the atrophy of skills/recall.
